### PR TITLE
🚧Build app settings system

### LINKED ISSUE #248: 🚧Build app settings system
### I have done the items below . . .

- [X] I have updated the title by replacing the '**_<title_**>' section.

### Description

Build an application settings system.  This will be used to turn on and off various application behaviors.  The system should automatically check if app settings exist.  If the application settings do not exist, the file will be auto-created and all application settings with default settings will be added to the file.

This should be set up as a service with the name `AppSettingsService` with an interface and full unit testing.

The settings will be simple key-value pairs in JSON format.
Create 2 app settings for the width and height of the application.  Refactor/add code to make use of these 2 settings.

Add a method overload to the `App` class of the method `CreateWindow()`.  This class will create an instance of the singleton app settings object and check if the settings exist. If the settings do not exist, use a default width and height of **1500 x 800**.
The default width and height should be constants in the `App` class.

Change the method used in the `Program.cs` file for the **VelaptorTesting** project to use the new `CreateWindow()` method overload.

### Acceptance Criteria

**This issue is finished when:**
- [x] The settings should be saved in JSON format
- [x] The check for file existence and creation with default settings should only be done one time upon creation of the service singleton.  This means once upon application startup.
    - If this was done every time, then that could be stress on the system by checking and loading file data.
- [x] ~The settings service should hold all of the app setting names and values as a `record` type.~
- [x] ~The settings service should have a method that returns the settings for evaluation and use by other parts of the application~
- [x] The settings service will be a singleton in the IoC container
- [x] Overload method named `CreateWindow()` added to the `App` class.
  - This method will pull the window width and height app settings for users to set the size of the application.
- [x] The app settings are set up to be used as default instead of using hard-coded values using the `CreateWindow(unit, unit)` method
- [x] Unit tests added
- [x] All unit tests pass

### ToDo Items

- [x] Draft pull request created and linked to this issue
- [X] Priority label added to issue (**_low priority_**, **_medium priority_**, or **_high priority_**)
- [x] Issue linked to the proper project
- [X] Issue linked to proper milestone

### Issue Dependencies

_No response_

### Related Work

- #247

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by CalvinWilkinson**
<h1 style="font-weight:bold" align="center">Preview Feature Pull Request</h1>
<h2 style="font-weight:bold" align="center">📋Review Checklist📋</h2>

- [x] The **_head(source)_** branch for this pull request is a **_preview feature_** branch, with the correct naming syntax, in the following example:
  - A **_head(source)_** branch is the branch being merged into a target branch.
    <details closed><summary>Example</summary>

      ``` xml
      Syntax: preview/feature/<issue-num>-<description>
      Example: preview/feature/123-my-preview-feature
      ```
    </details>

- [x] The **_head(source)_** branch for this pull request is created from a **_preview release_** branch, with the correct naming syntax, in the following example:
  <details closed><summary>Example</summary>

    ``` xml
    Syntax: preview/v<major>.<minor>.<patch>-preview.<prev-num>
    Example: preview/v1.2.3-preview.4
    ```
  </details>

- [x] The **_base(target)_** branch for this pull request is a **_preview release_** branch, with the correct naming syntax, in the following example:
  - A **_base(target)_** branch is the branch that the **_head(source)_** branch is merging into.

    <details closed><summary>Example</summary>

      ``` xml
      Syntax: preview/v<major>.<minor>.<patch>-preview.<prev-num>
      Example: preview/v1.2.3-preview.4
      ```
    </details>

💡For more information on branching, refer to the project's [branching documentation](../../Documentation/Branching.md).

- [x] Associated issue exists and is linked to this pull request.
  - One issue per pull request.

- [x] Pull request title matches the title of the associated issue.

- [x] A `preview` label has been added to the pull request.

- [x] The labels attached to this PR match the labels attached to the associated issue.

- [x] Project assigned matches the project of the associated issue.

- [x] Milestone assigned matches the milestone of the associated issue.

- [x] My code follows the coding style of this project.
  - The style is enforced by the **_.editorconfig_** files in the project and displayed as warnings.  If there is an edge case, reach out, and let's discuss it.

- [x] I have manually tested my code changes to the best of my ability.

- [x] All tests passed locally.
  - This is required because unit tests might pass locally but not in the CI environment during the status check process or vice-versa.
  - Tests might pass on the developer's machine but not necessarily on the code reviewer's machine.
  - Status checks run for this pull request when merging into the **_preview/v\*.\*.\*-preview.\*_** branch.  These status checks run every time a change is pushed to the pull request.  These checks validate version syntax, tagging, builds, unit tests, and more.

### CODE REVIEW

File: Velaptor/InternalExtensionMethods.cs [hunk 1]
```diff
@@ -177,6 +177,17 @@ public static string TrimDirSeparatorFromEnd(this string value)
             return value;
         }
 
+        /// <summary>
+        /// Converts the given <paramref name="path"/> to a cross platform path.
+        /// </summary>
+        /// <param name="path">The file or directory path.</param>
+        /// <returns>The cross platform version of the <paramref name="path"/>.</returns>
+        /// <returns>
+        ///     This just takes all '\' characters and changes them to '/' characters.
```
Comment (kselena) [inline Velaptor/InternalExtensionMethods.cs:186]: Grammar
```suggestion
        ///     This changes all '\' characters to '/' characters.
```

File: Velaptor/InternalExtensionMethods.cs [hunk 2]
```diff
@@ -177,6 +177,17 @@ public static string TrimDirSeparatorFromEnd(this string value)
...
+        ///     The '/' directory separator is valid on windows and linux systems.
```
Comment (kselena) [inline Velaptor/InternalExtensionMethods.cs:187]: Refactor total code base to capitalize Windows and Linux
```suggestion
        ///     The '/' directory separator is valid on Windows and Linux systems.
```

File: Velaptor/Services/AppSettingsService.cs [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+            catch (JsonException e)
+            {
+                var exMsg = $"There was an issue loading the application settings at the path '{this.appSettingsFilePath}'.";
+                exMsg += $"{Environment.NewLine}The file could be corrupt.";
+                throw new AppSettingsException(exMsg, e);
+            }
+        }
+
+        /// <summary>
+        /// Creates default settings and an associated file to store them.
```
Comment (kselena) [inline Velaptor/Services/AppSettingsService.cs:117]: Grammar
```suggestion
        /// Creates default settings and an associated file for storage.
```

File: Velaptor/Services/IAppSettingsService.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
+﻿// <copyright file="IAppSettingsService.cs" company="KinsonDigital">
+// Copyright (c) KinsonDigital. All rights reserved.
+// </copyright>
+
+namespace Velaptor.Services;
+
+/// <summary>
+/// Provides management of application settings.
```
Comment (kselena) [inline Velaptor/Services/IAppSettingsService.cs:8]: Sentence structure
```suggestion
/// Provides application settings management.
```

**Review by kselena — CHANGES_REQUESTED** [review]

**Review by kselena — APPROVED** [review]

**Review by KinsonDigitalAdmin — APPROVED** [review]

**Pull request merged**